src/common/iceberg.cpp

```diff
@@ -188,19 +188,32 @@ string GenerateMetaDataUrl(FileSystem &fs, const string &meta_path, string &tabl
 }
 
 
-string IcebergSnapshot::GetMetaDataPath(const string &path, FileSystem &fs, string metadata_compression_codec, string table_version = DEFAULT_VERSION_HINT_FILE, string version_format = DEFAULT_TABLE_VERSION_FORMAT) {
+string IcebergSnapshot::GetMetaDataPath(const string &path, FileSystem &fs, string metadata_compression_codec, string table_version = DEFAULT_TABLE_VERSION, string version_format = DEFAULT_TABLE_VERSION_FORMAT) {
+	string version_hint;
+	string meta_path = fs.JoinPath(path, "metadata");
 	if (StringUtil::EndsWith(path, ".json")) {
+		// We've been given a real metadata path. Nothing else to do.
 		return path;
-	}
-
-	auto meta_path = fs.JoinPath(path, "metadata");
-	string version_hint;
-		if(StringUtil::EndsWith(table_version, ".text")||StringUtil::EndsWith(table_version, ".txt")) {
-		version_hint = GetTableVersion(meta_path, fs, table_version);
-	} else {
+	} else if (!fs.DirectoryExists(meta_path)) {
+		// Make sure we have a metadata directory to look in
+		throw IOException("Cannot open \""+path+"\": Metadata directory does not exist");
```

Comment:
Bit of a nitpick, feel free to ignore this if there is no other reason to touch the branch

Formatting can be used within an exception message:
This could instead be:`"Cannot open \"%s\": Metadata directory does not exist", path`

Reply:
A good nitpick. I felt dirty as I wrote that but didn't go back to clean it up.